kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala

```diff
@@ -164,22 +170,15 @@ class ExecuteStatement(
   }
 
   override protected def runInternal(): Unit = {
-    if (shouldRunAsync) {
-      executeStatement()
-      val sessionManager = session.sessionManager
-      val asyncOperation = new Runnable {
-        override def run(): Unit = waitStatementComplete()
-      }
-      try {
-        val backgroundOperation =
-          sessionManager.submitBackgroundOperation(asyncOperation)
-        setBackgroundHandle(backgroundOperation)
-      } catch onError("submitting query in background, query rejected")
-    } else {
-      setState(OperationState.RUNNING)
-      executeStatement()
-      setState(OperationState.FINISHED)
-    }
+    executeStatement()
+    val sessionManager = session.sessionManager
+    val asyncOperation: Runnable = () => waitStatementComplete()
+    try {
+      val opHandle = sessionManager.submitBackgroundOperation(asyncOperation)
```

Comment:
can't we use the returned `futrue` directly rather than add a `CountDownLatch` ?
```
backgroundOperation.get(timeout)
```

Reply:
Let me try

Reply:
Updated